sklearn/utils/_tags.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+
+_DEFAULT_TAGS = {
+    'non_deterministic': False,
+    'requires_positive_X': False,
+    'requires_positive_y': False,
+    'X_types': ['2darray'],
+    'poor_score': False,
+    'no_validation': False,
+    'multioutput': False,
+    "allow_nan": False,
+    'stateless': False,
+    'multilabel': False,
+    '_skip_test': False,
+    '_xfail_checks': False,
+    'multioutput_only': False,
+    'binary_only': False,
+    'requires_fit': True,
+    'preserves_dtype': [np.float64],
+    'requires_y': False,
+    'pairwise': False,
+}
+
+
+def _safe_tags(estimator, key=None, default=None):
+    """Safely get estimator tags for common checks.
+
+    :class:`~sklearn.BaseEstimator` provides the estimator tags machinery.
+    However, if a compatible estimator does not inherit from this base class,
+    we should default to the default tag.
+
+    Parameters
+    ----------
+    estimator : estimator object
+        The estimator from which to get the tag.
+
+    key : str, default=None
+        Tag name to get. By default (`None`), all tags are returned.
+
+    default : list of {str, dtype} or bool, default=None
+        When `esimator.get_tags()` is not implemented, default` allows to
+        define the default value of a tag if it is not present in
+        `_DEFAULT_TAGS` or to overwrite the value in `_DEFAULT_TAGS` if it the
+        tag is defined. When `default is None`, no default values nor
+        overwriting will take place.
+
+    Returns
+    -------
+    tags : dict
+        The estimator tags.
+    """
+    if hasattr(estimator, "_get_tags"):
+        if key is not None:
+            try:
+                return estimator._get_tags().get(key, _DEFAULT_TAGS[key])
```

Comment:
Shouldn't this just be 
```suggestion
                return estimator._get_tags()[key]
```
?

Reply:
We could be extra tolerant if scikit-learn introduces a new tag in a new release so that the third party does not have to push a quick compat release just to avoid breaking compat with the new scikit-learn release by returning the new default value.

On the other hand it's a bit more magic. But maybe it's worth it in this case?

Reply:
this goes against our docs:

> Note however that all tags must be present in the dict
 
What bothers me here is that there's no difference anymore between `_get_tags()` and `_more_tags()` from a third-party point of view.